packages/quantic/force-app/main/default/lwc/quanticQueryError/quanticQueryError.js

```diff
@@ -0,0 +1,149 @@
+import goBack from '@salesforce/label/c.quantic_GoBack';
+import disconnectedException from '@salesforce/label/c.quantic_DisconnectedException';
+import noEndpointsException from '@salesforce/label/c.quantic_NoEndpointsException';
+import invalidTokenException from '@salesforce/label/c.quantic_InvalidTokenException';
+import organizationIsPausedException from '@salesforce/label/c.quantic_OrganizationIsPausedException';
+import disconnectedTitle from '@salesforce/label/c.quantic_DisconnectedTitle';
+import noEndpointsTitle from '@salesforce/label/c.quantic_NoEndpointsTitle';
+import invalidTokenTitle from '@salesforce/label/c.quantic_InvalidTokenTitle';
+import organizationIsPausedTitle from '@salesforce/label/c.quantic_OrganizationIsPausedTitle';
+import genericErrorTitle from '@salesforce/label/c.quantic_GenericErrorTitle';
+import disconnectedDesc from '@salesforce/label/c.quantic_DisconnectedDesc';
+import noEndpointsDesc from '@salesforce/label/c.quantic_NoEndpointsDesc';
+import invalidTokenDesc from '@salesforce/label/c.quantic_InvalidTokenDesc';
+import organizationIsPausedDesc from '@salesforce/label/c.quantic_OrganizationIsPausedDesc';
+import genericErrorDesc from '@salesforce/label/c.quantic_GenericErrorDesc';
+
+export default class QuanticQueryError extends LightningElement {
+  /** @type {string} */
+  @api engineId;
+
+  /** @type {import("coveo").QueryError} */
+  queryError;
+  /** @type {import("coveo").Unsubscribe} */
+  unsubscribe;
+
+  /**
+   * @type {string}
+   */
+  @track type;
+  /**
+   * @type {Boolean}
+   */
+  @track hasError;
+  /**
+   * @type {string}
+   */
+  @track error
+
+  showMoreInfo = false;
+
+  labels = {
+    coveoOnlineHelpLink,
+    moreInformation,
+    checkForMore,
+    community,
+    contactCoveoSupportTeam,
+    goBack
+  }
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+  }
+
+  /**
+   * @param {import("coveo").SearchEngine} engine
+   */
+  @api
+  initialize(engine) {
+    this.queryError = CoveoHeadless.buildQueryError(engine);
+    this.unsubscribe = this.queryError.subscribe(() => this.updateState());
+  }
+
+  disconnectedCallback() {
+    if (this.unsubscribe) {
+      this.unsubscribe();
+    }
+  }
+
+  updateState() {
+    this.type = this.queryError.state.error?.type;
+    this.hasError = this.queryError.state.hasError;
+    this.error = this.queryError.state.error ? JSON.stringify(this.queryError.state.error, null, 2): "";
+  }
+
+  get errorTitle() {
+    switch (this.type) {
+      case disconnectedException:
+        return disconnectedTitle;
+      case noEndpointsException:
+        return noEndpointsTitle;
+      case invalidTokenException:
+        return invalidTokenTitle;
+      case organizationIsPausedException:
+        return organizationIsPausedTitle;
+      default:
+        return genericErrorTitle;
+    }
```

Comment:
You could even make a map of `{ exception: { title, description } }` to not have to do a switch case every time?

```suggestion
    const title = errorMap[this.type]?.title || genericErrorTitle;
    return title;
```